pkg/broker/handler/processors/deliver/processor.go

```diff
@@ -41,6 +43,14 @@ type Processor struct {
 
 	// Targets is the targets from config.
 	Targets config.ReadonlyTargets
+
+	// RetryOnFailure if set to true, the processor will send the event
+	// to the retry topic if the delivery fails.
+	RetryOnFailure bool
+
+	// RetryEvents is the Pubsub protocol to send events
+	// to the retry topic.
+	RetryEvents *pubsub.Protocol
```

Comment:
Can we use CloudEvents pubsub Client instead of protocol?

Reply:
Why? I prefer protocol because it's more direct.

Reply:
My understanding is that Client is a higher level interface that allows you to only deal with Events while with protocol you still need to deal with `message`.

Reply:
I don't have a strong preference actually. How about let's defer this a bit and get this PR in first?

Reply:
It should be as simple as replace the protocol with client and call `client.Send(ctx, event)`.  If we don't do it now, I am afraid we may never get to it.

Examples in ingress:

https://github.com/google/knative-gcp/blob/master/pkg/broker/ingress/multi_topic_decouple_sink.go#L88-L89

https://github.com/google/knative-gcp/blob/master/pkg/broker/ingress/multi_topic_decouple_sink.go#L110-L123

Reply:
> we may never get to it

Maybe it's an indicator it doesn't make any difference?

Reply:
I think it's better to not get to the `bindings` concept if you don't need to. This makes the logic much easier to follow without understanding additional concepts. 

So both approach has the same effect (I believe the client implementation internally just converts events to message and calls protocol). But it's easier for someone else to read and maintain using a higher level abstraction.